### PR TITLE
Added signature validation section to install scripts docs

### PR DESCRIPTION
**Pull request by MichalPavlik**
## Summary

As our pipeline now supports signing the .sh version of the script, we should tell our customers how to get public key/signature and how to verify the script.

I will create this PR as a draft, because the hosting of the key and signature isn't resolved yet and links will be added or modified. It's still ready for review from the wording and styling perspective.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/tools/dotnet-install-script.md](https://github.com/dotnet/docs/blob/31495fb4d253d2f439fe0459bc3acd2c9de2c4ae/docs/core/tools/dotnet-install-script.md) | [dotnet-install scripts reference](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script?branch=pr-en-us-36841) |

### CODE REVIEW

File: docs/core/tools/dotnet-install-script.md [hunk 1]
```diff
@@ -305,6 +305,18 @@ Manually installing .NET doesn't add the environment variables system-wide, and
 
 There is no uninstall script. For information about manually uninstalling .NET, see [How to remove the .NET Runtime and SDK](../install/remove-runtime-sdk-versions.md#scripted-or-manual).
 
+## Signature validation of dotnet-install.sh
+
+Signature validation is an important security measure that helps ensure the authenticity and integrity of a script. By verifying the signature of a script, you can be sure that it has not been tampered with and that it comes from a trusted source.
+
+Here is a step-by-step guide on how to verify the authenticity of the `dotnet-install.sh` script using GPG:
+
+1. **Install GPG**: GPG (GNU Privacy Guard) is a free and open-source tool for encrypting and signing data. You can install it on your system using your package manager. For example, on Ubuntu or Debian, you can use the command `sudo apt-get install gnupg`.
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:314]:
```suggestion
1. **Install GPG**: GPG (GNU Privacy Guard) is a free and open-source tool for encrypting and signing data. You can install it by following the [instructions on the GPG website](https://gnupg.org/download/).
```

File: docs/core/tools/dotnet-install-script.md [hunk 2]
```diff
@@ -305,6 +305,18 @@ Manually installing .NET doesn't add the environment variables system-wide, and
...
+2. **Import our public key**: Download [public key](https://dot.net/v1/dotnet-install.asc) file, and you can then import it into your GPG keyring by running the command `gpg --import dotnet-install.asc`.
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:315]:
```suggestion
2. **Import our public key**: Download the install-scripts [public key](https://dot.net/v1/dotnet-install.asc) file, and then import it into your GPG keyring by running the command `gpg --import dotnet-install.asc`.
```

File: docs/core/tools/dotnet-install-script.md [hunk 3]
```diff
@@ -305,6 +305,18 @@ Manually installing .NET doesn't add the environment variables system-wide, and
...
+3. **Download the signature file**: The signature file for our bash script is available at `https://dot.net/v1/dotnet-install.sig`. You can download it using a tool like `wget` or `curl`.
+4. **Verify the signature**: To verify the signature of our bash script, run the command `gpg --verify dotnet-install.sig dotnet-install.sh`. This will check the signature of the `dotnet-install.sh` file against the signature in the `dotnet-install.sig` file.
+5. **Check the result**: If the signature is valid, you will see a message containing `Good signature from "Microsoft DevUXTeamPrague <devuxteamprague@microsoft.com>"`. This means that the script has not been tampered with and can be trusted.
+
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:319]: We should have an example terminal run-through here - using each command and showing the output.

**Review by baronfel — APPROVED** [review]

**Review by tdykstra — APPROVED** [review]
LGTM Merge after @baronfel's comments are addressed.

File: docs/core/tools/dotnet-install-script.md (hunk 3 above)
Comment (MichalPavlik) [reply docs/core/tools/dotnet-install-script.md:319]: I added examples, but I copied just output from `gpg`. Adding output from `apt install` and `wget` would be too verbose IMO.

File: docs/core/tools/dotnet-install-script.md [hunk 4]
```diff
@@ -305,6 +305,60 @@ Manually installing .NET doesn't add the environment variables system-wide, and
...
+2. **Import our public key**: Download the install-scripts [public key](https://dot.net/v1/dotnet-install.asc) file, and then import it into your GPG keyring by running the command `gpg --import dotnet-install.asc`.
+3. **Download the signature file**: The signature file for our bash script is available at `https://dot.net/v1/dotnet-install.sig`. You can download it using a tool like `wget` or `curl`.
+4. **Verify the signature**: To verify the signature of our bash script, run the command `gpg --verify dotnet-install.sig dotnet-install.sh`. This will check the signature of the `dotnet-install.sh` file against the signature in the `dotnet-install.sig` file.
+5. **Check the result**: If the signature is valid, you will see a message containing `Good signature from "Microsoft DevUXTeamPrague <devuxteamprague@microsoft.com>"`. This means that the script has not been tampered with and can be trusted.
+
+### Preparing environment
+
+Installing GPG and importing public key is one time operation.
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:322]:
```suggestion
Installing GPG and importing our public key is a one time operation.
```

File: docs/core/tools/dotnet-install-script.md [hunk 5]
```diff
@@ -305,6 +305,60 @@ Manually installing .NET doesn't add the environment variables system-wide, and
...
+
+```bash
+sudo apt install gpg
+wget https://dot.net/v1/dotnet-install.asc
+gpg --import dotnet-install.asc
+```
+
+Example of successful key import output:
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:330]:
```suggestion
When successful, you should see output like the following:
```

File: docs/core/tools/dotnet-install-script.md [hunk 6]
```diff
@@ -305,6 +305,60 @@ Manually installing .NET doesn't add the environment variables system-wide, and
...
+gpg: directory '/home/<user>/.gnupg' created
+gpg: keybox '/home/<user>/.gnupg/pubring.kbx' created
+gpg: /home/<user>/.gnupg/trustdb.gpg: trustdb created
+gpg: key B9CF1A51FC7D3ACF: public key "Microsoft DevUXTeamPrague <devuxteamprague@microsoft.com>" imported
+gpg: Total number processed: 1
+gpg:               imported: 1
+```
+
+### Download and verification
+
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:342]:
```suggestion

With the key imported, you can now download the script and the signature, then verify the script matches the signature:
```

File: docs/core/tools/dotnet-install-script.md [hunk 7]
```diff
@@ -305,6 +305,60 @@ Manually installing .NET doesn't add the environment variables system-wide, and
...
+```bash
+wget https://dot.net/v1/dotnet-install.sh
+wget https://dot.net/v1/dotnet-install.sig
+gpg --verify dotnet-install.sig dotnet-install.sh
+```
+
+Example of successful verification output:
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:349]:
```suggestion
When successful, you should see output like the following:
```

File: docs/core/tools/dotnet-install-script.md [hunk 8]
```diff
@@ -305,6 +305,60 @@ Manually installing .NET doesn't add the environment variables system-wide, and
...
+```
+gpg: Signature made <datetime>
+gpg:                using RSA key B9CF1A51FC7D3ACF
+gpg: Good signature from "Microsoft DevUXTeamPrague <devuxteamprague@microsoft.com>" [unknown]
+gpg: WARNING: This key is not certified with a trusted signature!
+gpg:          There is no indication that the signature belongs to the owner.
+Primary key fingerprint: 2B93 0AB1 228D 11D5 D7F6  B6AC B9CF 1A51 FC7D 3ACF
+```
+
+The warning means that you don't trust the public key in the keyring, but the script is still verified. Status code returned by the verification command should be `0`.
```
Comment (baronfel) [inline docs/core/tools/dotnet-install-script.md:360]:
```suggestion
The warning means that you don't trust the public key in the keyring, but the script is still verified. The exit code returned by the verification command should be `0`, indicating success.
```

**Review by baronfel — APPROVED** [review]
Thanks for the terminal examples @MichalPavlik - I made some minor grammar updates, but other than those I am very happy with this new content :)

**Pull request merged**